File: BookNestBackend/BookNest.Services/Database/Entities/Event.cs
using BookNest.Services.Database.Entities.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookNest.Services.Database.Entities
{
    public class Event
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(1000)]
        public string? Description { get; set; }

        [Required]
        public int EventCategoryId { get; set; }

        [ForeignKey(nameof(EventCategoryId))]
        public EventCategory EventCategory { get; set; } = null!;

        [Required]
        public int OrganizerId { get; set; }

        [ForeignKey(nameof(OrganizerId))]
        public Organizer Organizer { get; set; } = null!;

        [Required]
        public DateTime EventDate { get; set; }

        [Required]
        public TimeSpan EventTime { get; set; }

        [Required]
        [Column(TypeName = "nvarchar(50)")]
        public EventType EventType { get; set; }
        
        public string? Address { get; set; }
        
        public string? City { get; set; }

        public string? Country { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TicketPrice { get; set; }

        [Required]
        public int Capacity { get; set; }

        [Required]
        public bool IsActive { get; set; } = true;

        public string? ImageUrl { get; set; }

        [Required]
        public int ReservedSeats { get; set; }

        public ICollection<EventReservation> EventReservations { get; set; } = new List<EventReservation>();
        public ICollection<Review> Reviews { get; set; } = new List<Review>();
        public ICollection<Notification> Notifications { get; set; } = new List<Notification>();
        
    }
}

File: BookNestFrontend/booknest_desktop/lib/models/event.dart
class Event {
  final int id;
  final String name;
  final String? description;
  final int eventCategoryId;
  final String eventCategoryName;
  final String organizerName;
  final DateTime eventDate;
  final String eventTime;
  final String eventType;
  final String? address;
  final String? city;
  final String? country;
  final double ticketPrice;
  final int capacity;
  final bool isActive;
  final String? imageUrl;
  final int reservedSeats;

  Event({
    required this.id,
    required this.name,
    this.description,
    required this.eventCategoryId,
    required this.eventCategoryName,
    required this.organizerName,
    required this.eventDate,
    required this.eventTime,
    required this.eventType,
    this.address,
    this.city,
    this.country,
    required this.ticketPrice,
    required this.capacity,
    required this.isActive,
    this.imageUrl,
    required this.reservedSeats,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'],
      name: json['name'] ?? '',
      description: json['description'],
      eventCategoryId: json['eventCategoryId'] ?? 0,
      eventCategoryName: json['eventCategoryName'] ?? '',
      organizerName: json['organizerName'] ?? '',
      eventDate: DateTime.tryParse(json['eventDate']?.toString() ?? '') ?? DateTime.now(),
      eventTime: json['eventTime'] ?? '00:00:00',
      eventType: json['eventType'] ?? '',
      address: json['address'],
      city: json['city'],
      country: json['country'],
      ticketPrice: (json['ticketPrice'] as num?)?.toDouble() ?? 0,
      capacity: (json['capacity'] as num?)?.toInt() ?? 0,
      isActive: json['isActive'] ?? false,
      imageUrl: json['imageUrl'],
      reservedSeats: (json['reservedSeats'] as num?)?.toInt() ?? 0,
    );
  }
}
